### PR TITLE
feat: give access to logged in user to customization context

### PR DESCRIPTION
**Pull request by romain-gilliotte**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)
- [ ] Ensure that Types have been updated according to your changes (if needed)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/agent/src/agent/utils/query-string.ts [hunk 1]
```diff
@@ -126,7 +127,7 @@ export default class QueryStringParser {
       throw new ValidationError(`Invalid timezone: "${timezone}"`);
     }
 
-    return timezone;
+    return { ...context.state.user, timezone };
```
Comment (Scra3) [inline packages/agent/src/agent/utils/query-string.ts:130]: we should to add a test please :) I know you love it.

File: packages/agent/src/agent/routes/access/csv-related.ts [hunk 2]
```diff
@@ -27,19 +32,19 @@ export default class CsvRelatedRoute extends RelationRoute {
...
-      );
-
-    const gen = CsvGenerator.generate(projection, header, filter, this.foreignCollection, list);
+    const gen = CsvGenerator.generate(
+      caller,
+      projection,
+      header,
+      filter,
+      this.foreignCollection,
+      async (rec: Caller, fil: PaginatedFilter, proj: Projection) =>
```
Comment (Scra3) [inline packages/agent/src/agent/routes/access/csv-related.ts:45]: call instead rec ?

File: packages/agent/test/agent/routes/access/chart.test.ts [hunk 3]
```diff
@@ -96,13 +96,8 @@ describe('ChartRoute', () => {
...
 
       expect(dataSource.getCollection('books').aggregate).toHaveBeenCalledWith(
-        {
-          conditionTree: null,
-          search: null,
-          searchExtended: false,
-          segment: null,
-          timezone: 'Europe/Paris',
-        },
+        { timezone: 'Europe/Paris' },
```
Comment (Scra3) [inline packages/agent/test/agent/routes/access/chart.test.ts:99]: it is a little bit long, but can we add a user in the body for all the tests please :).

File: packages/datasource-toolkit/test/decorators/rename/collection.test.ts [hunk 4]
```diff
@@ -186,8 +196,18 @@ describe('RenameCollectionDecorator', () => {
...
-      const rows = await newPersons.aggregate(personsPaginatedFilter, aggregate, null);
-      expect(persons.aggregate).toHaveBeenCalledWith(personsPaginatedFilter, aggregate, null);
+      const rows = await newPersons.aggregate(
+        factories.caller.build(),
+        personsPaginatedFilter,
+        aggregate,
+        null,
+      );
+      expect(persons.aggregate).toHaveBeenCalledWith(
+        factories.caller.build(),
```
Comment (Scra3) [inline packages/datasource-toolkit/test/decorators/rename/collection.test.ts:206]: you should compare to the caller instance passed in the aggregate method.
I have added `renderingId: Math.random()` in the factory and there are 65  tests failed.

File: packages/agent/src/agent/routes/access/csv-related.ts (hunk 2 above)
Comment (jeffladiray) [reply packages/agent/src/agent/routes/access/csv-related.ts:45]: (Same goes for `fil` & `proj` :))

**Review by Scra3 — APPROVED** [review]

**Pull request merged**